src/native/managed/cdacreader/src/Contracts/Registry.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+internal sealed class Registry
+{
+    // Contracts that have already been created for a target.
+    // Items should not be removed from this, only added.
+    private readonly Dictionary<Type, IContract> _contracts = [];
```

Comment:
Is there a concurrency issue here?

Reply:
I think currently no, legacy DAC single-threads everything with the `DAC_ENTER`/`DAC_LEAVE` macros on the C++ side:

https://github.com/dotnet/runtime/blob/776ff7e40c784634f9c9f51a36fd5dec47a0064f/src/coreclr/debug/daccess/dacimpl.h#L3800-L3820